vllm/v1/core/scheduler.py

```diff
@@ -424,13 +433,33 @@ def update_from_output(
                 # This must be called before me make the EngineCoreOutput.
                 stopped = self._check_stop(request)
 
+                # Extract sample logprobs if needed.
+                logprobs_token_ids: List[torch.Tensor] = []
+                logprobs: List[torch.Tensor] = []
+                if request.sampling_params.logprobs:
+                    assert logprobs_token_ids_cpu is not None
+                    assert logprobs_cpu is not None
+                    # Here we assume there is 1 generated token per step.
+                    logprobs_token_ids = [logprobs_token_ids_cpu[req_index]]
+                    logprobs = logprobs_cpu[req_index]
+
+                # Extract prompt logprobs for this req if needed.
+                # FIXME(rob): handle partial request. Currently we throw away
```

Comment:
@afeldman-nm - we can handle this by adding `prompt_logprobs` to `Request` and updating the state as we go